cluster-autoscaler/cloudprovider/clusterapi/clusterapi_scalableresource.go

```diff
@@ -60,3 +67,25 @@ type scalableResource interface {
 	InstanceGPUCapacity() (resource.Quantity, error)
 	InstanceMaxPodsCapacity() (resource.Quantity, error)
 }
+
+func unmarkMachineForDeletion(controller *machineController, machine *Machine) error {
+	u, err := controller.dynamicclient.Resource(*controller.machineResource).Namespace(machine.Namespace).Get(context.TODO(), machine.Name, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
+	if u == nil {
+		return fmt.Errorf("unknown machine %s", machine.Name)
+	}
+
+	u = u.DeepCopy()
```

Comment:
Is the dynamicclient caching? I don't _think_ it is, and if not, we don't need to do a copy

Reply:
i'm not sure about the caching, i copied this code from the mark function. i will do some investigation and it is not caching, i'll remove this copy. thanks!

Reply:
i looked into this a little further, and there is a [cached discovery client](https://github.com/kubernetes/client-go/blob/master/discovery/cached/disk/cached_discovery.go#L273), but [i don't think we are using it](https://github.com/openshift/kubernetes-autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_provider.go#L160).

i will remove the deep copy

Reply:
It's actually the dynamic client we are using for this (see line 72) which is a different thing, but it also isn't cached :)

Reply:
oh yeah, i misclicked. meant to link this one:

https://github.com/openshift/kubernetes-autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_provider.go#L150